src/GGLSetup.py

```diff
@@ -101,12 +114,53 @@ def install_greengrass_lite_from_source(commit_id: str, region: str):
 
         # run nucleus
         os.chmod('misc/run_nucleus', 0o775)
-        process = subprocess.run(['sudo', './misc/run_nucleus'],
-                                 check=True,
-                                 text=True,
-                                 stdout=subprocess.PIPE,
-                                 stderr=subprocess.PIPE)
-        print("Successfully installed the nucleus from source")
+        try:
+            process = subprocess.run(['sudo', './misc/run_nucleus'],
+                                     check=True,
+                                     text=True,
+                                     stdout=subprocess.PIPE,
+                                     stderr=subprocess.PIPE)
+            print("Successfully installed the nucleus from source")
+        except subprocess.CalledProcessError as e:
+            print(f"FATAL: run_nucleus failed with exit code {e.returncode}")
+            print(f"stdout: {e.stdout}")
+            print(f"stderr: {e.stderr}")
+            raise Exception(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.



Similar issue at line numbers 142 and 160.